### DISCUSSION TITLE
Provide option for border on left or right of tree to separate it from other open windows.

### DISCUSSION THREAD

**Discussion by jeffschwartz** [nvim-tree/nvim-tree.lua#1583] [Q&A]
**Describe the solution you'd like**

Provide an option that would turn on a one sided border whose color is also configurable. The border should be placed on the right if nvim-tree is configured to open on the left and the reverse if nvim-tree is configured to open on the right.

Perhaps a border implementation could take advantage of Neovim's `WinSeparator` which provides similar capabilities to splits.

Another option, or perhaps in addition, would be to provide for either a lighter or darker background for the tree.

This request is not just for aesthetics, but would help serve those  whose eyesight is poor and would make it easier to see where the tree ends and other windows/splits begin,  such as when using the mouse to widen the viewport of the tree.

Thanks to all those involved with this plugin.

**Comment by alex-courtis** [comment]
> help serve those whose eyesight is poor and would make it easier to see where the tree ends and other windows/splits begin

Fanstastic.

Take a look at `:help nvim-tree-highlight`

Try this to demonstrate some options:
```vim
hi NvimTreeWinSeparator guibg=green
hi NvimTreeNormal guibg=red
hi NvimTreeEndOfBuffer guibg=blue
```

**Reply by jeffschwartz** [reply]
Thank you  @alex-courtis. I sincerely apologize for not  having thoroughly read the documentation prior to creating the original issue.

I have added the three highlight groups that you identified to my init.vim and it works as stated. My setting are as follows:

```vim
hi NvimTreeWinSeparator guibg=white
hi NvimTreeNormal guibg=none
hi NvimTreeEndOfBuffer guibg=none
```
As I  haven't found a highlight group relevant to the actual width of the separator, is it possible or would it be possible to have the separator's style and specifically its width conform to the same style that `highlight WinSeparator guibg=None` provides?

Again, I am really sorry about my OP. As an OS contributor I should have known better to post a question prior to have thoroughly read the documentation.

**Reply by alex-courtis** [reply]
> As I haven't found a highlight group relevant to the actual width of the separator,

A quick :help hasn't revealed any means of changing the width of vim separators. `:help fcs` allows you to change the characters but only allows single.

> Again, I am really sorry about my OP.

No apologies expected or needed.

**Comment by jeffschwartz** [comment]
I discovered that 
```vim
set laststatus=3
highlight winseparator guibg=none
```
is working but that many color schemes aren't taking advantage of it. For instance, if you install TokyoNight and if you look really, really hard you can see a faint dark border separating nvim-tree from other windows. In contrast, Gruvbox displays a very distinct and pronounced border around nvim-tree. So, in summary. I don't think that this is an nvim-tree issue at all. We will have to wait for all the color scheme authors who support Neovim to catch up.

**Reply by alex-courtis** [reply]
> is working but that many color schemes aren't taking advantage of it. For instance, if you install TokyoNight and if you look really, really hard you can see a faint dark border separating nvim-tree from other windows. In contrast, Gruvbox displays a very distinct and pronounced border around nvim-tree. So, in summary. I don't think that this is an nvim-tree issue at all. We will have to wait for all the color scheme authors who support Neovim to catch up.

Could this be a race in plugin loading order? You can check the actual state of the highlight groups via `:so /usr/share/nvim/runtime/syntax/hitest.vim`
